### PR TITLE
Update PSP support feature and add documentation

### PR DESCRIPTION
**Pull request by almaslennikov**
Addressed post-merge code review feedback and updated README.md with the documentation about the feature

Signed-off-by: Alexander Maslennikov <amaslennikov@nvidia.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -286,6 +287,9 @@ spec:
 
 Can be found at: `mellanox.com_v1alpha1_hostdevicenetwork_cr.yaml`
 
+## Pod Security Policy
+Network-operator supports [Pod Security Policies](https://kubernetes.io/docs/concepts/policy/pod-security-policy/). When deployed with `psp.enabled=True`, privileged PSP is created and applied to all network-operator's pods. Requires [admission controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#how-do-i-turn-on-an-admission-control-plug-in) to be enabled.
```
Comment (adrianchiris) [inline README.md:291]: as this README is not related to Helm, small nit:

> When deployed with `psp.enabled=True`

When NicClusterPolicy is created with  `psp.enabled=True`

**Review by adrianchiris — DISMISSED** [review]
Thanks for addressing my comments @almaslennikov !

a nit to fix in README, other than that its LGTM

**Review by e0ne — APPROVED** [review]
